### PR TITLE
Refactor Gitimporter using fetchcode

### PR DESCRIPTION
**Pull request by ziadhany**
Reference:  #806
Signed-off-by: ziad <ziadhany2016@gmail.com>

### CODE REVIEW

**Comment by TG1999** [comment]
@ziadhany , please add logs for gitlab importer.

File: vulnerabilities/tests/test_data_source.py
Comment (TG1999) [inline vulnerabilities/tests/test_data_source.py:44]: Thanks for removing old tests, IMO we should have some new tests to check the new GitImporter

**Review by TG1999 — CHANGES_REQUESTED** [review]

**Comment by ziadhany** [comment]
> @ziadhany , please add logs for gitlab importer.


```
Importing data using vulnerabilities.importers.gitlab.GitLabGitImporter
Unknow package type 'conan'
Unknow package type 'conan'
...
Unknow package type 'conan'
Unknow package type 'conan'
Unknow package type 'go'
....
Unknow package type 'go'
Unknow package type 'go'
....
parse_yaml_file: affected_range is not parsable: '(,)' type:'nuget' error: InvalidVersion("'None' is not a valid <class 'univers.versions.NugetVersion'>")
 Traceback (most recent call last):
  File "/home/ziad/PycharmProjects/vulnerablecode/vulnerabilities/importers/gitlab.py", line 223, in parse_gitlab_advisory
    affected_version_range = vrc.from_native(affected_range)
  File "/home/ziad/PycharmProjects/vulnerablecode/venv/lib/python3.10/site-packages/univers/version_range.py", line 616, in from_native
    VersionConstraint(comparator="=", version=cls.version_class(str(lower_bound)))
  File "<attrs generated init univers.versions.NugetVersion>", line 7, in __init__
    self.__attrs_post_init__()
  File "/home/ziad/PycharmProjects/vulnerablecode/venv/lib/python3.10/site-packages/univers/versions.py", line 87, in __attrs_post_init__
    raise InvalidVersion(f"{self.string!r} is not a valid {self.__class__!r}")
univers.versions.InvalidVersion: 'None' is not a valid <class 'univers.versions.NugetVersion'>

parse_yaml_file: affected_range is not parsable: '[4.0.0],(,3.4.0),[4.0.0,4.1.2)' type:'nuget' error: VersionRangeParseError('Ranges overlap: [4.0.0],(,3.4.0),[4.0.0,4.1.2)')
 Traceback (most recent call last):
  File "/home/ziad/PycharmProjects/vulnerablecode/vulnerabilities/importers/gitlab.py", line 223, in parse_gitlab_advisory
    affected_version_range = vrc.from_native(affected_range)
  File "/home/ziad/PycharmProjects/vulnerablecode/venv/lib/python3.10/site-packages/univers/version_range.py", line 605, in from_native
    restrictions = maven.VersionRange(string).restrictions
  File "/home/ziad/PycharmProjects/vulnerablecode/venv/lib/python3.10/site-packages/univers/maven.py", line 226, in __init__
    raise VersionRangeParseError("Ranges overlap: %s" % spec)
univers.maven.VersionRangeParseError: Ranges overlap: [4.0.0],(,3.4.0),[4.0.0,4.1.2)

parse_yaml_file: affected_range is not parsable: '[2.2],[2.1]' type:'nuget' error: VersionRangeParseError('Ranges overlap: [2.2],[2.1]')
 Traceback (most recent call last):
  File "/home/ziad/PycharmProjects/vulnerablecode/vulnerabilities/importers/gitlab.py", line 223, in parse_gitlab_advisory
[GitLabImporterLogs.txt](https://github.com/nexB/vulnerablecode/files/9639764/GitLabImporterLogs.txt)

    affected_version_range = vrc.from_native(affected_range)
  File "/home/ziad/PycharmProjects/vulnerablecode/venv/lib/python3.10/site-packages/univers/version_range.py", line 605, in from_native
    restrictions = maven.VersionRange(string).restrictions
  File "/home/ziad/PycharmProjects/vulnerablecode/venv/lib/python3.10/site-packages/univers/maven.py", line 226, in __init__
    raise VersionRangeParseError("Ranges overlap: %s" % spec)
univers.maven.VersionRangeParseError: Ranges overlap: [2.2],[2.1]
.....

Successfully imported data using vulnerabilities.importers.gitlab.GitLabGitImporter
```

**Comment by ziadhany** [comment]
> @ziadhany , please add logs for gitlab importer.


```
Importing data using vulnerabilities.importers.gitlab.GitLabGitImporter
Unknow package type 'conan'
Unknow package type 'conan'
...
Unknow package type 'conan'
Unknow package type 'conan'
Unknow package type 'go'
....
Unknow package type 'go'
Unknow package type 'go'
....
parse_yaml_file: affected_range is not parsable: '(,)' type:'nuget' error: InvalidVersion("'None' is not a valid <class 'univers.versions.NugetVersion'>")
 Traceback (most recent call last):
  File "/home/ziad/PycharmProjects/vulnerablecode/vulnerabilities/importers/gitlab.py", line 223, in parse_gitlab_advisory
    affected_version_range = vrc.from_native(affected_range)
  File "/home/ziad/PycharmProjects/vulnerablecode/venv/lib/python3.10/site-packages/univers/version_range.py", line 616, in from_native
    VersionConstraint(comparator="=", version=cls.version_class(str(lower_bound)))
  File "<attrs generated init univers.versions.NugetVersion>", line 7, in __init__
    self.__attrs_post_init__()
  File "/home/ziad/PycharmProjects/vulnerablecode/venv/lib/python3.10/site-packages/univers/versions.py", line 87, in __attrs_post_init__
    raise InvalidVersion(f"{self.string!r} is not a valid {self.__class__!r}")
univers.versions.InvalidVersion: 'None' is not a valid <class 'univers.versions.NugetVersion'>

parse_yaml_file: affected_range is not parsable: '[4.0.0],(,3.4.0),[4.0.0,4.1.2)' type:'nuget' error: VersionRangeParseError('Ranges overlap: [4.0.0],(,3.4.0),[4.0.0,4.1.2)')
 Traceback (most recent call last):
  File "/home/ziad/PycharmProjects/vulnerablecode/vulnerabilities/importers/gitlab.py", line 223, in parse_gitlab_advisory
    affected_version_range = vrc.from_native(affected_range)
  File "/home/ziad/PycharmProjects/vulnerablecode/venv/lib/python3.10/site-packages/univers/version_range.py", line 605, in from_native
    restrictions = maven.VersionRange(string).restrictions
  File "/home/ziad/PycharmProjects/vulnerablecode/venv/lib/python3.10/site-packages/univers/maven.py", line 226, in __init__
    raise VersionRangeParseError("Ranges overlap: %s" % spec)
univers.maven.VersionRangeParseError: Ranges overlap: [4.0.0],(,3.4.0),[4.0.0,4.1.2)

parse_yaml_file: affected_range is not parsable: '[2.2],[2.1]' type:'nuget' error: VersionRangeParseError('Ranges overlap: [2.2],[2.1]')
 Traceback (most recent call last):
  File "/home/ziad/PycharmProjects/vulnerablecode/vulnerabilities/importers/gitlab.py", line 223, in parse_gitlab_advisory
    affected_version_range = vrc.from_native(affected_range)
  File "/home/ziad/PycharmProjects/vulnerablecode/venv/lib/python3.10/site-packages/univers/version_range.py", line 605, in from_native
    restrictions = maven.VersionRange(string).restrictions
  File "/home/ziad/PycharmProjects/vulnerablecode/venv/lib/python3.10/site-packages/univers/maven.py", line 226, in __init__
    raise VersionRangeParseError("Ranges overlap: %s" % spec)
univers.maven.VersionRangeParseError: Ranges overlap: [2.2],[2.1]
.....

Successfully imported data using vulnerabilities.importers.gitlab.GitLabGitImporter
```

File: vulnerabilities/importers/gitlab.py
Comment (TG1999) [inline vulnerabilities/importers/gitlab.py:74]: @ziadhany this will change the qualified name for importer and the databases will still have the old qualified name stored in them, which will now not be recogonised by improver, either write a migration to support this change or do not change the name of importer.

File: vulnerabilities/tests/test_data_source.py
Comment (TG1999) [reply vulnerabilities/tests/test_data_source.py:44]: @ziadhany gentle ping!

Comment (ziadhany) [reply vulnerabilities/tests/test_data_source.py:44]: Sorry I was sick 🤧, I'm working on it

**Review by TG1999 — APPROVED** [review]
LGTM! Thanks

**Comment by TG1999** [comment]
Please rebase with the latest main

File: vulnerabilities/tests/test_data_source.py
Comment (pombredanne) [inline vulnerabilities/tests/test_data_source.py:34]: IMHO this would be best moved to a new test_oval.py

File: vulnerabilities/tests/test_data_source.py
Comment (pombredanne) [inline vulnerabilities/tests/test_data_source.py:287]: IMHO this would be best moved to a new test_oval.py and not just deleted

File: vulnerabilities/tests/test_data_source.py
Comment (pombredanne) [inline vulnerabilities/tests/test_data_source.py:28]: Does this do a live clone over the network?

**Review by pombredanne — COMMENTED** [review]
Thanks!
I posted a few comments inline to ensure we do drop tests. Beside these, this looks good.

File: vulnerabilities/tests/test_data_source.py
Comment (ziadhany) [reply vulnerabilities/tests/test_data_source.py:28]: yes, it clones the repository
( I just mock a constructor and then make the Git importer run, so I can test the result )

File: vulnerabilities/tests/test_data_source.py
Comment (pombredanne) [reply vulnerabilities/tests/test_data_source.py:287]: I cannot approve this if we just delete all these tests above.

Comment (TG1999) [reply vulnerabilities/tests/test_data_source.py:287]: I have secured oval importer tests in the file itself https://github.com/nexB/vulnerablecode/pull/817/files#diff-eb1cffdadef3eacc470900468af6cce0bdd6ae476bc9ae94ce512863c060f850R40, and just deleted the GitImporter tests that are not related with the latest importers.

**Review by pombredanne — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**
